presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionProcess.java

```diff
@@ -112,9 +112,13 @@ public NativeExecutionProcess(
     /**
      * Starts the external native execution process. The method will be blocked by connecting to the native process's /v1/info endpoint with backoff retries until timeout.
      */
-    public void start()
+    public synchronized void start()
             throws ExecutionException, InterruptedException, IOException
     {
+        if (process != null && process.isAlive()) {
+            return;
+        }
+
         String executablePath = getProcessWorkingPath(SystemSessionProperties.getNativeExecutionExecutablePath(session));
```

Comment:
If we are going to share native process among queries, we cannot use session properties to launch it.

Reply:
> If we are going to share native process among queries, we cannot use session properties to launch it.

True. But we only share native process among queries in test scenario, for most of the cases they should be using the same session properties. In the case test case requires using different session properties, we can turn off the process sharing.